crates/matrix-sdk-base/src/store/memory_store.rs

```diff
@@ -77,13 +78,14 @@ pub struct MemoryStore {
             HashMap<(String, Option<String>), HashMap<OwnedEventId, HashMap<OwnedUserId, Receipt>>>,
         >,
     >,
+    media: StdRwLock<RingBuffer<(OwnedMxcUri, String /* unique key */, Vec<u8>)>>,
```

Comment:
Is the MXC also a unique key? (so we could get rid of the `String`)

Reply:
Two media can have the same URI but with a different format for example. So both `MxcUri` and `String` (the unique key value) are necessary.